src/python/pants/backend/python/util_rules/local_dists_pep660.py

```diff
@@ -0,0 +1,539 @@
+        if line.startswith("Version: "):
+            version = line[len("Version: "):].strip()
+            break
+    # standardize the name of the dist-info directory
+    old_metadata_path = metadata_path
+    metadata_path = pkg + "-" + version + ".dist-info"
+    shutil.move(os.path.join(build_dir, old_metadata_path), os.path.join(build_dir, metadata_path))
+
+# Any RECORD* file will be incorrect since we are creating the wheel.
+for file in os.listdir(os.path.join(build_dir, metadata_path)):
+    if file == "RECORD" or file.startswith("RECORD."):
+        os.unlink(os.path.join(build_dir, metadata_path, file))
+metadata_wheel_file = os.path.join(build_dir, metadata_path, "WHEEL")
+if not os.path.exists(metadata_wheel_file):
+    with open(metadata_wheel_file, "w") as f:
+        f.write('''\
+Wheel-Version: 1.0
+Generator: pantsbuild
+Root-Is-Purelib: true
+Tag: {{}}
+Build: 0.editable
+'''.format(tags))
+
+if direct_url:
+    # We abuse pex to get PEP660 editable wheels installed in a virtualenv.
+    # Pex and pip do not know that this is an editable install.
+    # We can't add direct_url.json to the wheel, because that has to be added
+    # by the wheel installer. So, we will rename this file once installed.
+    direct_url_file = os.path.join(build_dir, metadata_path, "direct_url__pants__.json")
+    with open(direct_url_file , "w") as f:
+        f.write('''\
+{{{{
+    "url": "{{}}",
+    "dir_info": {{{{
+        "editable": true
+    }}}}
+}}}}
+'''.format(direct_url))
+
+pth_file_arcname = pkg + "__pants__.pth"
+
+_record = []
+def record(file_path, file_arcname):
+    with open(file_path, "rb") as f:
+        file_digest = hashlib.sha256(f.read()).digest()
+    file_hash = "sha256=" + base64.urlsafe_b64encode(file_digest).decode().rstrip("=")
+    file_size = str(os.stat(file_path).st_size)
+    _record.append(",".join([file_arcname, file_hash, file_size]))
+
+wheel_path = "{{}}-{{}}-0.editable-{{}}.whl".format(pkg, version, tags)
+with zipfile.ZipFile(os.path.join(dist_dir, wheel_path), "w") as whl:
+    record(pth_file_path, pth_file_arcname)
+    whl.write(pth_file_path, pth_file_arcname)
+    # based on wheel.wheelfile.WheelFile.write_files (by @argonholm MIT license)
+    for root, dirnames, filenames in os.walk(os.path.join(build_dir, metadata_path)):
+        dirnames.sort()
+        for name in sorted(filenames):
+            path = os.path.normpath(os.path.join(root, name))
+            if os.path.isfile(path):
+                arcname = os.path.relpath(path, build_dir).replace(os.path.sep, "/")
+                record(path, arcname)
+                whl.write(path, arcname)
+    record_path = os.path.join(metadata_path, "RECORD")
+    _record.extend([record_path + ",,", ""])  # "" to add newline at eof
+    whl.writestr(record_path, os.linesep.join(_record))
+
+print("editable_path: {{editable_path}}".format(editable_path=wheel_path))
+"""
+
+
+def interpolate_backend_wrapper(  # based on interpolate_backend_shim
```

Comment:
noisy comment. (I do appreciate the origin information, we could have that as review comments, though, to keep the code clean ;) )